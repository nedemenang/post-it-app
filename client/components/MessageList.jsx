import React, { Component } from 'react';
import { List, Subheader } from 'material-ui';
import lodash from 'lodash';
import '../public/style.scss';
import Message from './Message';

/**
 * @class MessageList
 * @extends {Component}
 */
class MessageList extends Component {

  /**
   * Creates an instance of MessageList.
   * @param {object} props props object
   * @memberof MessageList
   */
  constructor(props) {
    super(props);
    this.state = {};
  }

  /**
   *
   * Renders message list page
   * @returns {JSX} returns message list page
   * @memberof MessageList
   */
  render() {
    let groupsname = '';
    let groupId = '';
    if (!lodash.isEmpty(this.props.selectedGroup)) {
      groupsname = this.props.selectedGroup.groupName;
      groupId = this.props.selectedGroup.groupId;
    }
    return (
      <div className="bottomMargin">
        <List>
        <Subheader><strong>{groupsname} Message List </strong></Subheader>
        {
            this.props.messages.map((message, i) =>
            <Message userRead = {this.props.userReadMessages}
                  groupId={groupId} message={message} key={i} />)
        }
        </List>
      </div>
    );
  }
}

export default MessageList;
