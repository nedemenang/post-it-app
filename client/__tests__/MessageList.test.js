import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MessageList from '../components/MessageList.jsx';

describe('the message list component', () => {
  let mountedComponent;

  const str = location.href;
  const str2 = str.replace('#', ' ');

  const loggedInUser = {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    isAuthenticated: true,
    phoneNo: null,
    profilePic: `${str2}static/files/blank-profile-pic.png`
  };

  const group = { groupId: 'TUsUaucS5nb5kuOP6SpvTUcX9v3',
    groupName: 'Test Group',
    newMessage: false };

  const messages = [{ messageBody: 'Test message',
    postedBy: 'testUser@gmail.com',
    postedByDisplayName: 'Test User',
    postedon: '',
    priority: 'Normal',
    profilePic: `${str2}static/files/blank-profile-pic.png` },
  { messageBody: 'Test message',
    postedBy: 'testUser@gmail.com',
    postedByDisplayName: 'Test User',
    postedon: '',
    priority: 'Normal',
    profilePic: `${str2}static/files/blank-profile-pic.png` },
  { messageBody: 'Test message',
    postedBy: 'testUser@gmail.com',
    postedByDisplayName: 'Test User',
    postedon: '',
    priority: 'Normal',
    profilePic: `${str2}static/files/blank-profile-pic.png` }];

  const messageListItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<MessageList messages = {messages}
      selectedGroup={group} usersReadMessages={[]}
      loggedInUser = {loggedInUser}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = messageListItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = messageListItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });


  it('should recieve props', () => {
    expect(Object.keys(messageListItem().props()).length).toBeGreaterThan(0);
    expect(messageListItem().instance().props.selectedGroup).toEqual(group);
    expect(messageListItem().instance().props.loggedInUser)
    .toEqual(loggedInUser);
    expect(messageListItem().instance().props.messages).toEqual(messages);
  });
});
