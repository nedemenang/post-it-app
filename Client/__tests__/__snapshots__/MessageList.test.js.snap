// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the message list component should match snapshot test 1`] = `
<div
  className="bottomMargin"
>
  <List>
    <Subheader
      inset={false}
    >
      <strong>
         Message List 
      </strong>
    </Subheader>
    <Message
      groupId="TUsUaucS5nb5kuOP6SpvTUcX9v3"
      message={
        Object {
          "messageBody": "Test message",
          "postedBy": "testUser@gmail.com",
          "postedByDisplayName": "Test User",
          "postedon": "",
          "priority": "Normal",
          "profilePic": "about:blankstatic/files/blank-profile-pic.png",
        }
      }
    />
    <Message
      groupId="TUsUaucS5nb5kuOP6SpvTUcX9v3"
      message={
        Object {
          "messageBody": "Test message",
          "postedBy": "testUser@gmail.com",
          "postedByDisplayName": "Test User",
          "postedon": "",
          "priority": "Normal",
          "profilePic": "about:blankstatic/files/blank-profile-pic.png",
        }
      }
    />
    <Message
      groupId="TUsUaucS5nb5kuOP6SpvTUcX9v3"
      message={
        Object {
          "messageBody": "Test message",
          "postedBy": "testUser@gmail.com",
          "postedByDisplayName": "Test User",
          "postedon": "",
          "priority": "Normal",
          "profilePic": "about:blankstatic/files/blank-profile-pic.png",
        }
      }
    />
  </List>
  <p
    className="error"
  />
</div>
`;
